megameklab/src/megameklab/ui/dialog/PrintQueueDialog.java

```diff
@@ -265,10 +270,34 @@ protected void okButtonActionPerformed(ActionEvent evt) {
     }
 
     private void saveUnitList() {
-        // todo: Refactor saveListFile to be a static method of some util class
-        // --Pavel
+        var entities = getEntities();
+        if (entities.isEmpty()) {
+            return;
+        }
+
         linkForce();
-        new ClientGUI(new Client("", "", 0), null).saveListFile(getEntities(), "Units");
+
+        var fileChooser = new JFileChooser(".");
+        fileChooser.setDialogTitle(Messages.getString("ClientGUI.saveUnitListFileDialog.title"));
+        var filter = new FileNameExtensionFilter(
+                Messages.getString("ClientGUI.descriptionMULFiles"), CG_FILEPATHMUL);
+        fileChooser.setFileFilter(filter);
+        fileChooser.setSelectedFile(new File(Strings.isNotBlank(mulFileName) ? mulFileName : entities.get(0).getShortName() + " etc." + CG_FILEPATHMUL));
+
+        if (!(fileChooser.showSaveDialog(parent) == JFileChooser.APPROVE_OPTION) || fileChooser.getSelectedFile() == null) {
+            return;
+        }
+
+        File file = fileChooser.getSelectedFile();
+        if (!FilenameUtils.getExtension(file.getName()).equalsIgnoreCase(CG_FILEPATHMUL)) {
+            file = new File(file + "." + CG_FILEPATHMUL);
+        }
+
+        try {
+            EntityListFile.saveTo(file, entities);
+        } catch (IOException e) {
+            JOptionPane.showMessageDialog(getFrame(), "Error: " + e.getMessage());
```

Comment:
Would suggest using the MMLogger with (exception, Message, title) if this is going to be kept.

Also suggest using String.format() for the message.

Reply:
Honestly would suggest String.format() on any string concatenation.

Reply:
Done